storage/rocksdb/ha_rocksdb.cc

```diff
@@ -11973,23 +11973,30 @@ void rocksdb_set_collation_exception_list(THD *const thd,
   *static_cast<const char **>(var_ptr) = val;
 }
 
-void rocksdb_set_bulk_load(THD *const thd, struct st_mysql_sys_var *const var
-                                               MY_ATTRIBUTE((__unused__)),
-                           void *const var_ptr, const void *const save) {
-  Rdb_transaction *&tx = get_tx_from_thd(thd);
+int rocksdb_check_bulk_load(THD *const thd,
+                            struct st_mysql_sys_var *var
+                                MY_ATTRIBUTE((__unused__)),
+                            void *save,
+                            struct st_mysql_value *value) {
+    Rdb_transaction *&tx = get_tx_from_thd(thd);
 
-  if (tx != nullptr) {
-    const int rc = tx->finish_bulk_load();
-    if (rc != 0) {
-      // NO_LINT_DEBUG
-      sql_print_error("RocksDB: Error %d finalizing last SST file while "
-                      "setting bulk loading variable",
-                      rc);
-      abort_with_stack_traces();
+    if (tx != nullptr) {
+        const int rc = tx->finish_bulk_load();
+        if (rc != 0) {
+            // NO_LINT_DEBUG
+            sql_print_error(
+                "RocksDB: Error %d finalizing last SST file while "
+                "setting bulk loading variable",
+                rc);
+            THDVAR(thd, bulk_load) = 0;
+            return 1;
+        }
     }
-  }
 
-  *static_cast<bool *>(var_ptr) = *static_cast<const bool *>(save);
+    long long intbuf;
+    value->val_int(value, &intbuf);
```

Comment:
The test rocksdb_sys_vars.bulk_load_basic is failing to use "on" as 1. I think since it's a bool, value could be a string and needs to evaluate "ON", "OFF" to 1 and 0 too.

Reply:
Really? This is weird, the test succeeds for me in a debug build. I assumed that the ON/OFF/YES/NO bool transformation was already done before coming into the check. Debugging a bit into it and yeah, you are correct, it is coming in to the check function as a string type when ON/OFF/YES/NO and not an integer type, the integer transformation just converts the string to '0' and does not recognize the ON/YES as being worth '1'. This is a good catch and in fact makes me think we have a bug over in TokuDB where it is using a similar technique to 'fail' a SET. I'll build up a fix and update when I have re-pushed.

Reply:
OK, changes made, re-tested and pushed. Thanks!